### ISSUE TITLE
Identify Go binaries with custom magics

### ISSUE THREAD

**Issue by mr-tz** [mandiant/flare-floss#840]
Thank you for letting me know, I was not aware! 

I took a look at the identification used for go just now, it may be nice to extend the Go detections to include binaries with custom magics. This can be the case for garble obfuscated Go binaries. Here https://github.com/mandiant/flare-floss/blob/master/floss/language/identify.py#L85

Using strings from the linked go runtime may be good to signature off of in addition, or perhaps signature scans on runtime code for each arch x version. A strings based approach may be sufficient though.

_Originally posted by @stevemk14ebr in https://github.com/mandiant/flare-floss/issues/830#issuecomment-1639029967_

**Comment by stevemk14ebr** [comment]
I suspect a first pass improvement would be quite simple. A string based approach using strings we already signature on for capa is a great start: https://github.com/stevemk14ebr/capa-rules/blob/6822e748a97bb8b8cd91627fb7a46280193f3222/compiler/go/compiled-with-go.yml

Additional testing against `garble` obfuscator in particular is necessary. I would potentially suggest the following as well in case the prior is insufficient:
```
runtime.morestack
runtime.morestack_noctxt
runtime.newproc
runtime.deferproc
runtime.gcWriteBarrier
runtime.Gosched
```

These are core internals that are present in all binaries I've tested, obfuscated and non.

**Comment by stevemk14ebr** [comment]
PR by Sara https://github.com/mandiant/flare-floss/pull/908!

**Comment by mr-tz** [comment]
closed via #908 - please reopen if there's other things we should add
